File: CS-Examples/07_DocumentOperation/LinkHeadersFooters.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;

namespace LinkHeadersFooters
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
           // Create a new instance of the Document class.
			Document srcDoc = new Document();

			// Load the source document from a file using a relative path.
			srcDoc.LoadFromFile(@"..\..\..\..\..\..\..\Data\Template_N1.docx");

			// Create another instance of the Document class.
			Document dstDoc = new Document();

			// Load the destination document from a file using a relative path.
			dstDoc.LoadFromFile(@"..\..\..\..\..\..\..\Data\Template_N2.docx");

			// Link the header of the first section in the source document to the previous section's header.
			srcDoc.Sections[0].HeadersFooters.Header.LinkToPrevious = true;

			// Link the footer of the first section in the source document to the previous section's footer.
			srcDoc.Sections[0].HeadersFooters.Footer.LinkToPrevious = true;

			// Iterate through each section in the source document.
			foreach (Section section in srcDoc.Sections)
			{
				// Clone each section and add it to the destination document.
				dstDoc.Sections.Add(section.Clone());
			}

			// Specify the output file name.
			string output = "LinkHeadersFooters_out.docx";

			// Save the destination document to a file with the specified file format (Docx2013).
			dstDoc.SaveToFile(output, FileFormat.Docx2013);

			// Dispose of the source and destination documents to free up resources.
			srcDoc.Dispose();
			dstDoc.Dispose();

            //Launching the document
            WordDocViewer(output);
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB-Examples/07_DocumentOperation/LinkHeadersFooters.vb
Imports Spire.Doc
Imports Spire.Doc.Documents

Namespace LinkHeadersFooters
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Create a new instance of the Document class and assign it to the srcDoc variable.
			Dim srcDoc As New Document()

			'Load a Word document from the specified file path and assign it to the srcDoc variable.
			srcDoc.LoadFromFile("..\..\..\..\..\..\..\Data\Template_N1.docx")

			'Create a new instance of the Document class and assign it to the dstDoc variable.
			Dim dstDoc As New Document()

			'Load a Word document from the specified file path and assign it to the dstDoc variable.
			dstDoc.LoadFromFile("..\..\..\..\..\..\..\Data\Template_N2.docx")

			'Set the LinkToPrevious property of the header in the first section of srcDoc to True.
			srcDoc.Sections(0).HeadersFooters.Header.LinkToPrevious = True

			'Set the LinkToPrevious property of the footer in the first section of srcDoc to True.
			srcDoc.Sections(0).HeadersFooters.Footer.LinkToPrevious = True

			'Iterate through each section in srcDoc and add a cloned section to dstDoc.
			For Each section As Section In srcDoc.Sections
				dstDoc.Sections.Add(section.Clone())
			Next section

			'Specify the output file name as "LinkHeadersFooters_out.docx".
			Dim output As String = "LinkHeadersFooters_out.docx"

			'Save the dstDoc document to the specified file path in the Docx2013 file format.
			dstDoc.SaveToFile(output, FileFormat.Docx2013)

			'Dispose of the resources used by srcDoc.
			srcDoc.Dispose()

			'Dispose of the resources used by dstDoc.
			dstDoc.Dispose()

			'Launching the document
			WordDocViewer(output)
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
